# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: caps/BasePrincipal.cpp
# Commit: 39f2e21623aa
# Full Hash: 39f2e21623aa7c19517e32cbd97dc1c3584c58ca
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2020-06-02 03:24:46
# Regressor Bug: 1638358
# File Overlap Count: 1
# Description:
#   Bug 1638358 - Cookie Schemeful Same-Site - part 5 - schemeful comparison, r=mayhemer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D75629
# ==============================================================================

diff -r 2c873c72bf1f -r 39f2e21623aa caps/BasePrincipal.cpp
--- a/caps/BasePrincipal.cpp	Mon Jun 01 16:49:05 2020 +0000
+++ b/caps/BasePrincipal.cpp	Mon Jun 01 16:49:03 2020 +0000
@@ -856,6 +856,19 @@
 }
 
 NS_IMETHODIMP
+BasePrincipal::GetScheme(nsACString& aScheme) {
+  aScheme.Truncate();
+
+  nsCOMPtr<nsIURI> prinURI;
+  nsresult rv = GetURI(getter_AddRefs(prinURI));
+  if (NS_FAILED(rv) || !prinURI) {
+    return NS_OK;
+  }
+
+  return prinURI->GetScheme(aScheme);
+}
+
+NS_IMETHODIMP
 BasePrincipal::SchemeIs(const char* aScheme, bool* aResult) {
   *aResult = false;
   nsCOMPtr<nsIURI> prinURI;